src/sonic-yang-mgmt/_sonic_yang_ext.py

```diff
@@ -0,0 +1,616 @@
+"""
+def loadYangModel(self):
+
+    try:
+        yangDir = self.yang_dir
+        self.yangFiles = glob(yangDir +"/*.yang")
+        for file in self.yangFiles:
+            if (self.load_schema_module(file) == False):
+                return False
+
+        # keep only modules name in self.yangFiles
+        self.yangFiles = [f.split('/')[-1] for f in self.yangFiles]
+        self.yangFiles = [f.split('.')[0] for f in self.yangFiles]
+        print(self.yangFiles)
+
+        # load json for each yang model
+        self.loadJsonYangModel()
+        # create a map from config DB table to yang container
+        self.createDBTableToModuleMap()
+
+    except Exception as e:
+        print("Yang Models Load failed")
+        raise e
+
+    return True
+
+"""
+load JSON schema format from yang models
+"""
+def loadJsonYangModel(self):
+
+    for f in self.yangFiles:
+        m = self.ctx.get_module(f)
+        if m is not None:
+            xml = m.print_mem(ly.LYD_JSON, ly.LYP_FORMAT)
+            self.yJson.append(parse(xml))
+
+    return
+
+"""
+Create a map from config DB tables to container in yang model
+"""
+def createDBTableToModuleMap(self):
+
+    for j in self.yJson:
+        # get module name
+        moduleName = j['module']['@name']
+        if "sonic-head" in moduleName or "sonic-common" in moduleName:
+            continue;
+        # get all top level container
+        topLevelContainer = j['module']['container']
+        if topLevelContainer is None:
+            raise Exception("topLevelContainer not found")
+
+        assert topLevelContainer['@name'] == moduleName
+
+        container = topLevelContainer['container']
+        # container is a list
+        if isinstance(container, list):
+            for c in container:
+                self.confDbYangMap[c['@name']] = {
+                    "module" : moduleName,
+                    "topLevelContainer": topLevelContainer['@name'],
+                    "container": c
+                    }
+        # container is a dict
+        else:
+            self.confDbYangMap[container['@name']] = {
+                "module" : moduleName,
+                "topLevelContainer": topLevelContainer['@name'],
+                "container": container
+                }
+    return
+
+"""
+Get module, topLevelContainer and json container for a config DB table
+"""
+def get_module_top_container(self, table):
+    cmap = self.confDbYangMap
+    m = cmap[table]['module']
+    t = cmap[table]['topLevelContainer']
+    c = cmap[table]['container']
+    return m, t, c
+
+"""
+Crop config as per yang models
+"""
+def cropConfigDB(self, croppedFile=None):
+
+    for table in self.jIn.keys():
+        if table not in self.confDbYangMap:
+            del self.jIn[table]
+
+    if croppedFile:
+        with open(croppedFile, 'w') as f:
+            dump(self.jIn, f, indent=4)
+
+    return
+
+"""
+Extract keys from table entry in Config DB and return in a dict
```

Comment:
Can you give an example?

Reply:
Thats why, I mentioned in  logs: https://github.com/Azure/SONiC/blob/master/doc/mgmt/SONiC_YANG_Model_Guidelines.md.

All these constraints are from Guidline, I can give reference of this doc in code, whereever we have such constraints. Thx.

Reply:
I will add below in code as well:

For Example: regex = <vlan_name>|<ip-prefix> and tableKey = "Vlan111|2a04:5555:45:6709::1/64"

1.) first code will extract key list from regex, i.e. vlan_name and ip_prefix.
2.) then will create another regex(regexV) to extract Values from tableKey by replacing "<key> --> extractor i.e. (.*?)" in regex.
3.) Then will extract values from tableKey with regexV.
4.) Resulting Dict will be: 
KeyDict = {"vlan_name": "Vlan111", "ip-prefix": "2a04:5555:45:6709::1/64"}